File: src/main/java/com/github/miyashy/karatesample/infrastructure/TodoRepositoryImpl.java
package com.github.miyashy.karatesample.infrastructure;

import com.github.miyashy.karatesample.domain.Todo;
import com.github.miyashy.karatesample.domain.repository.TodoRepository;
import org.springframework.stereotype.Repository;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@Repository
public class TodoRepositoryImpl implements TodoRepository {
    private final Map<UUID, Todo> map = new HashMap<>();

    @Override
    public Todo save(Todo todo) {
        map.put(todo.getId(), todo);
        return todo;
    }

    @Override
    public Todo find(UUID id) {
        return map.get(id);
    }

    @Override
    public void delete(Todo todo) {
        map.remove(todo.getId());
    }
}

File: src/main/java/com/github/miyashy/karatesample/infrastructure/TodoRepositoryImpl.kt
package com.github.miyashy.karatesample.infrastructure

import com.github.miyashy.karatesample.domain.Todo
import com.github.miyashy.karatesample.domain.repository.TodoRepository
import org.springframework.stereotype.Component
import java.util.*

@Component
class TodoRepositoryImpl: TodoRepository {
    val map: MutableMap<UUID, Todo> = mutableMapOf()
    override fun save(todo: Todo): Todo {
        map[todo.id] = todo
        return todo
    }

    override fun find(id: UUID): Todo? {
        return map[id]
    }

    override fun delete(todo: Todo) {
        map.remove(todo.id)
    }
}